connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -245,34 +245,19 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
             ConfigDef basicConfigDef = (connector instanceof SourceConnector)
                                        ? SourceConnectorConfig.configDef()
                                        : SinkConnectorConfig.configDef();
-            Map<String, ConfigValue> validatedConnectorConfig = validateBasicConnectorConfig(
-                    connector,
-                    basicConfigDef,
-                    connectorConfig
-            );
-            configValues.addAll(validatedConnectorConfig.values());
-            configKeys.putAll(basicConfigDef.configKeys());
-            allGroups.addAll(basicConfigDef.groups());
-
-            ConnectorConfig connectorConfigToEnrich = (connector instanceof SourceConnector)
-                    ? new SourceConnectorConfig(plugins(), connectorConfig)
-                    : new SinkConnectorConfig(plugins(), connectorConfig);
-            final ConfigDef connectorConfigDef = connectorConfigToEnrich.enrich(
+
+            ConfigDef newDef = new ConfigDef(basicConfigDef);
```

Comment:
This doesn't seem to be used anywhere? What is this accomplishing with the `embed`?